aten/src/ATen/native/ReplicationPadding.cpp

```diff
@@ -339,7 +342,11 @@ void replication_pad2d_out_cpu_template(Tensor& output,
   int dimslices = 0;
   int64_t nbatch = 1;
 
-  TORCH_CHECK(input_.numel() > 0 && (input_.dim() == 3 || input_.dim() == 4),
+  // allow 0 dim batch size and nothing else.
+  bool valid_dims = input_.size(1) != 0 && input_.size(2) != 0;
```

Comment:
for 3d tensor dims 0,1,2 should be non-zero, for 4d tensor dims 1,2,3 should be nonzero